gensim/models/base_any2vec.py

```diff
@@ -1172,24 +1198,46 @@ def _log_progress(self, job_queue, progress_queue, cur_epoch, example_count, tot
         trained_word_count : int
             Number of effective words used in training until now (after ignoring unknown words and trimming
             the sentence length).
+        total_samples : int
+            Number of effective samples used in training until now (differs from total_examples for Skip-Gram)
         elapsed : int
             Elapsed time since the beginning of training in seconds.
 
         """
+        if total_samples == 0:
+            loss = -1
+        else:
+            loss = self.get_latest_training_loss() / total_samples
         if total_examples:
             # examples-based progress %
-            logger.info(
-                "EPOCH %i - PROGRESS: at %.2f%% examples, %.0f words/s, in_qsize %i, out_qsize %i",
-                cur_epoch + 1, 100.0 * example_count / total_examples, trained_word_count / elapsed,
-                utils.qsize(job_queue), utils.qsize(progress_queue)
-            )
+            if self.compute_loss:
```

Comment:
can you fully refactor this function please (make it clearer & shorter, not `if { if { } else { } } else { if { } else { } } `)?

Hint
- generate pattern (logging template first)
- collect needed parameters to list/tuple
- use `*my_parameters`